sdk/eventhub/azure-eventhub-checkpointstoretable/tests/test_storage_table_partition_manager.py

```diff
@@ -1,8 +1,119 @@
 import pytest
 from azure.eventhub.extensions.checkpointstoretable import TableCheckpointStore
+from azure.data.tables import TableServiceClient
+import uuid
+import warnings
+import random
+import os
 
+STORAGE_CONN_STR = [os.environ.get("AZURE_STORAGE_CONN_STR", "Azure Storage Connection String")]
 
-def test_constructor():
-    client = TableCheckpointStore()
-    assert client is not None
-    
+def get_live_storage_table_client(storage_connection_str):
+    try:
+        table_name = chr(random.randint(ord('a'), ord('z'))) +(uuid.uuid4().hex)
+        table_service_client = TableServiceClient.from_connection_string(storage_connection_str)
+        table_service_client.create_table_if_not_exists(table_name)
+        return storage_connection_str, table_name
+    except:
+        pytest.skip("Storage table client can't be created")
+
+def remove_live_storage_table_client(storage_connection_str, table_name):
+    try:
+        table_service_client = TableServiceClient.from_connection_string(storage_connection_str)
+        table_service_client.delete_table(table_name)
+    except:
+        warnings.warn(UserWarning("storage table teardown failed"))
+
+def _claim_and_list_ownership(storage_connection_str, table_name):
```

Comment:
This test makes sure that you can claim ownership if an ownership doesn't already exist. Can you also add a test to ensure claim ownership works if an ownership does exist?

I think it would also be good to add a test that ensures that if your etag doesn't match what Tables has, then you get an `OwnershipLost` error. The 2 ways that can be triggered are:
1. Try to claim_ownership with an etag specified but the ownership doesn't exist in the Table. (This covers the case where something deleted the ownership that you're trying to claim)
2. Try to claim_ownership with a different etag than what exists in the Table. (This covers the case where another consumer claims ownership before you have a chance to)

Reply:
yes that sounds good to add.